### DISCUSSION TITLE
Pushbutton on Analog input set as Button not triggering

### DISCUSSION THREAD

**Discussion by kelvinyap9** [shanteacontrols/OpenDeck#317] [Q&A] [answered]
https://github.com/user-attachments/assets/f583b980-6fdc-4c58-96de-a804b00127af

Pushbutton connected to 4067 first mux pin (aka Analog0) 
Analog0 from mux -> pull-down resistor to GND -> Pushbutton -> 3.3V rail.

When Analog0 is set to Note, it can detect the button press and release, and triggers a MIDI input for both press and release.

When changing Analog to Message type to Button, then going into Button block, it's supposed to show up as Button->Analog->16, but no input is detected. Last 2 seconds of the video I am pressing the pushbutton multiple times.

I have also tried connecting as:

Analog0 from mux -> pull-up resistor to 3.3V -> Pushbutton -> GND.

Any advice appreciated!

**Comment by paradajz** [comment, marked as answer]
Fixed on master branch.

**Reply by mrbensinclair** [reply]
I'm having the same problem using firmware 7.12.0 running RPi Pico. If I configure the analog input as anything other than Button, it works fine, but when I set it to Button it does nothing.

**Reply by paradajz** [reply]
How did you wire up the button?

**Reply by mrbensinclair** [reply]
Pin 1 to analog in 0
Pin 1 to 10k resistor then to ground
Pin 2 to 3.3v supply

Enabled the analogue input, and selected button from the options on the related analog trigger.

Selected the appropriate button from the analog section on the button page.

Set trigger the CC93, value 127.

**Reply by paradajz** [reply]
Fixed. https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.12.1

**Reply by mrbensinclair** [reply]
Thanks. I can see it's working now, however I think it's too sensitive - I'm getting quite erratic output from that button. I have it set to latching, CC/0 off, CC=93, Value=64. Sometimes it works as expected (toggling between CC93=64, and CC93=0 with each successive press), but most of the time it sends several on/off commands at once. In the screenshot, that's the result of one single button press
<img width="608" height="380" alt="Screenshot 2026-01-05 at 10 46 14 AM" src="https://github.com/user-attachments/assets/0c9c0000-1a39-493f-ba55-f0768effb56e" />

I'm using a momentary footswitch, and I can see that if I set the analog input to CC instead of button, when I press the button it results in a range of CC values (see second screenshot)
<img width="609" height="558" alt="Screenshot 2026-01-05 at 10 51 00 AM" src="https://github.com/user-attachments/assets/2c509274-a529-4b4a-9599-fd4c78d3cea2" />

**Comment by paradajz** [comment]
Can you try with the attached firmware? I've tweaked the thresholds again (compressed since I can't upload uf2 directly).

[merged.zip](https://github.com/user-attachments/files/24430333/merged.zip)

**Reply by mrbensinclair** [reply]
That's a little better, but still a bit erratic. I'm still getting multiple on/off messages per button press about 60% of the time.

**Reply by paradajz** [reply]
What kind of button are you using?

**Reply by mrbensinclair** [reply]
A soft touch momentary footswitch. 

![image](https://github.com/user-attachments/assets/71c8cdae-1005-4c57-a3e1-0a783f6f1d23)

**Reply by paradajz** [reply]
Hm. Can you try this one?
[merged.zip](https://github.com/user-attachments/files/24431846/merged.zip)

**Reply by paradajz** [reply]
Also try comparing it with this one:
[merged.zip](https://github.com/user-attachments/files/24431985/merged.zip)

The upper one should probably work fine but will cause more RAM usage and I'd like to avoid that (since it's going to pass through button debouncer and I need to allocate more space for the debouncer then - analog inputs configured as digital inputs aren't run through any debouncer other than through on/off thresholds). The second one passes through full analog debouncer so I don't know if the response time will be good enough for you.

**Comment by paradajz** [comment]
I plan on releasing this one - I've spent some time reworking the button/analog debouncing mechanism so I'm fairly sure you'll have no problems with this.

[merged.zip](https://github.com/user-attachments/files/24437277/merged.zip)

**Reply by mrbensinclair** [reply]
Thank you. The third one is the most reliable. 

All three improved the performance, but I could still generate erratic messages using the first the first two if I pressed the button very slowly.

In the configurator, the analog block lights up green periodically (only when one of the analog inputs is set to button) when using both the first and third firmwares, but not the second one. This doesn't seem to be an issue (in that it doesn't generate any extra messages), but I thought it might be useful feedback.

Thank you so much for putting in so much time to make this work. It's very much appreciated!

**Reply by paradajz** [reply]
If I understood correctly, the last one causes no issues?

> In the configurator, the analog block lights up green periodically (only when one of the analog inputs is set to button) when using both the first and **third** firmwares, but not the second one.

Hm yeah this is a bit tricky without introducing more variables but I think I've managed to do it. If you'd like test it and let me know.

[merged.zip](https://github.com/user-attachments/files/24440385/merged.zip)

**Reply by paradajz** [reply]
But to clarify, once the button on analog input is pressed, configurator **should** indicate both the analog and button event. That is by design. The firmware above merely fixes an issue where the analog filter sends out messages through the system constantly during one button state, thus blinking several times. This doesn't cause an issue to the button processing since the same state sent again by the analog module will be ignored anyways, but still, it's not really how it should be, hence the fix above.

**Reply by paradajz** [reply]
> The firmware above merely fixes an issue where the analog filter sends out messages through the system constantly during one button state, thus blinking several times.

You can see this for yourself actually if you enable the system messages in the configurator.

**Reply by mrbensinclair** [reply]
Yes that is correct - the third one you sent causes no issues.

The new one works perfectly- consistent button behaviour, and the analog block only flashes when the button is pressed.

I had been viewing the system messages in the configurator throughout - it was super handy to isolate the problem.

**Reply by paradajz** [reply]
Great, thanks for the feedback. Sometimes it's hard to test certain issues all by myself.

New release is out:
https://github.com/shanteacontrols/OpenDeck/releases/tag/v7.12.2
